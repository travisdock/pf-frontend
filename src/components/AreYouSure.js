import React from 'react';
import Popup from "reactjs-popup";

class AreYouSure extends React.Component {
    render() {
        return (
            <Popup
            open={this.props.open === "ask"}
            closeOnDocumentClick
            onClose={this.props.closeModal}
            onOpen={this.props.onOpen}
            >
            <div className="table-popup">
                <p>{this.props.message}</p>
                <button className="delete" onClick={this.props.handleDelete} >Delete</button>
                <button className="close" onClick={this.props.closeModal} >Close</button>
            </div>
            </Popup>
        )
    }
};

export default AreYouSure;