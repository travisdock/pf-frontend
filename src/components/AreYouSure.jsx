import React from 'react';
import Popup from "reactjs-popup";

class AreYouSure extends React.Component {
    render() {
        return (
            <Popup
            open={this.props.open === "ask"}
            closeOnDocumentClick
            onClose={this.props.closeModal}
            onOpen={this.props.askIfSure}
            >
            <div className="table-popup">
                <p>Are You Sure?</p>
                <button onClick={this.props.handleDelete} >Delete</button>
                <button onClick={this.props.closeModal} >Close</button>
            </div>
            </Popup>
        )
    }
};

export default AreYouSure;