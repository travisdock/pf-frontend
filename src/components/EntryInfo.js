import React from 'react';
import Popup from "reactjs-popup";

class EntryInfo extends React.Component {
    render() {
        return (
            <Popup
            open={this.props.open === "info"}
            closeOnDocumentClick
            onClose={this.props.closeModal}
            >
            <div className="table-popup">
                <p>Amount: {"$" + Number(this.props.amount).toFixed(2)}</p>
                <p>Category: {this.props.category_name}</p>
                <p>Date: {this.props.date}</p>
                <p>Notes: {this.props.notes}</p>
                <button className="delete" onClick={this.props.askIfSure}>Delete Entry</button>
                <button className="update" onClick={this.props.openUpdateModal}>Update Entry</button>
                <button className="copy" onClick={() => this.props.copyEntry(this.props)}>Copy Entry</button>
            </div>
            </Popup>
        )
    }
};

export default EntryInfo;