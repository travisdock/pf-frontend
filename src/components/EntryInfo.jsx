import React from 'react';
import Popup from "reactjs-popup";

class EntryInfo extends React.Component {
    render() {
        return (
            <Popup
            open={this.props.open === "info"}
            closeOnDocumentClick
            onClose={this.props.closeModal}
            >
            <div className="table-popup">
                <p>Amount: {"$" + Number(this.props.amount).toFixed(2)}</p>
                <p>Category: {this.props.category_name}</p>
                <p>Date: {this.props.date}</p>
                <p>Notes: {this.props.notes}</p>
                <button onClick={this.props.askIfSure}>Delete Entry</button>
                <button onClick={this.props.openUpdateModal}>Update Entry</button>
            </div>
            </Popup>
        )
    }
};

export default EntryInfo;