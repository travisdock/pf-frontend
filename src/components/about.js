import React from 'react';

const About = () => (
  <div className="about">
    <h1>Sum.</h1>
    <p>Some guy once said something like "What gets measured, gets managed". For people having trouble managing their personal finances, simply measuring their income and expenses is the first step. Often just the act of knowing where your money is going brings about the motivation to make changes. Sum. is a very simple app that gives the user a lot of freedom in simply measuring. Create any number of custom categories for income and expenses. Then watch where the money goes. Once you actually see how your expenses add up it will almost be harder not to make any changes.</p>
  </div>
)

export default About;
