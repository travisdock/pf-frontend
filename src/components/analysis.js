import React from 'react';
import { connect } from 'react-redux';

import Chart from './chart';

class Analysis extends React.Component {
  state = {
    userEntries: []
  }

  componentDidMount() {
    const url = `http://localhost:3001/api/v1/month_category/${this.props.current_user.user_id}`
    fetch(url)
      .then(resp => resp.json())
      .then(resp => this.setState({userData: resp}))
  }
  render(){

    return(
      <div>{this.state.userData ?
        <div><Chart /></div>
        :
        <div>Loading...</div>
      }</div>
    )
  }
}

const mapStateToProps = state => ({
  current_user: state.current_user
});

export default connect(mapStateToProps)(Analysis);
