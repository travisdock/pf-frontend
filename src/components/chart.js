import React from 'react';
import {bb} from "billboard.js";
import { connect } from 'react-redux';

class Chart extends React.Component {

  state = {
    charts: {},
    currentChart: 0,
    stats: {},
    extraMonthInfo: {},
    load: false
  }

  componentDidMount() {
    const id = this.props.current_user.user_id

    fetch(`${process.env.REACT_APP_API}/api/v1/charts/${id}`)
      .then(resp => resp.json())
      .then(resp => {if (resp.error) {
        this.setState({error: resp.error})
      } else {
        const titles = Object.keys(resp.charts)
        const lastMonth = titles[titles.length - 1]
        this.setState({
          charts: resp.charts,
          stats: resp.stats,
          load: true,
          currentChart: resp.charts[lastMonth], currentTitle: lastMonth
        }, this.renderPieChart)
      }}
        )
  }

  selectChangePieChart = (e) => {
    const title = e.target.value
    this.setState( () => {
      return {
        currentChart: this.state.charts[title],
        currentTitle: title
      }
    }, this.renderPieChart)
  }

  renderPieChart() {
    const { currentChart, currentTitle } = this.state
    // const { currentTitle } = Object.keys(currentChart)[0]
    // debugger;
    bb.generate({
      data: {
        columns: currentChart,
        type: "pie",
        // onclick: function(d, element) { debugger }
      },
      title: {
        text: currentTitle,
        position: "top-center",
        padding: {
            top: 10,
           right: 10,
            bottom: 10,
            left: 10
        },
      },
      legend: {
        show: true,
        // contents: {
        //     bindto: "#legend",   // <ul id='legend'></ul>
        //     template: "<li style='background-color:#1f77b4'>data1</li>"
        //     or using function
        //     template: function(id, color, data) {
        //          // if you want omit some legend, return falsy value
        //          if (title !== "data1") {
        //               return "<li style='background-color:"+ color +">"+ title +"</li>";
        //          }
        //     }
        // },
        position: "bottom",  // bottom, right, inset
        equally: false,
        padding: 10,
        item: {
            // onclick: function(id) { debugger },
            // onover: function(id) { debugger },
            // onout: function(id) { debugger },
    
            // set tile's size
            tile: {
                width: 20,
                height: 15
            }
        }
      },
      bindto: "#chart"
    });
  }

  render() {
    return (
      <div className="content">
        {this.state.load ?
            <div className="chart-content">
              <select
                name="chart"
                value={this.state.currentTitle} //current chart month
                onChange={this.selectChangePieChart}
                >
                {Object.keys(this.state.charts).map(title => <option value={title} key={title} > {title} </option>) }
              </select>
              <div id="chart" />
            </div> :
            this.state.error ? <div>{this.state.error}</div> : <div>Loading...</div>
        }
      </div>
    )
  }
}
const mapStateToProps = state => ({
  current_user: state.current_user
});

export default connect(mapStateToProps)(Chart)



// This is the profit loss chart. I got rid of it because it was ugly but I may want to remake it one day so here is the code.
// renderPLChart() {
//   const { currentChart } = this.state
//   bb.generate({
//     data: {
//       x: "x",
//       columns: currentChart,
//     type: "area-spline"
//     },
//     axis: {
//       x: {
//         type: "timeseries",
//         tick: {
//           format: "%b"
//         }
//       }
//     },
//     bindto: "#chart"
//   });
// }