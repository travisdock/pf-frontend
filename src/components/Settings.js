import React from 'react';
import { connect } from 'react-redux';
import { updateUser } from '../actions';

// Import modals
import AreYouSure from './AreYouSure'
import UpdateCategory from './UpdateCategory'

import {
    handleUserUpdate, handleDelete,
    openModal, handleCategoryUpdate
} from './helpers/settingsHelpers';

import {
    closeModal, askIfSure,
    openUpdateModal, handleChange
} from './helpers/modalHelpers'

import {
    toggleIncome, toggleUntracked
} from './helpers/inputFormHelpers'

class Settings extends React.Component {
    constructor() {
        super();
        this.state = {
            open: false,
            form: {
                name: '',
                income: '',
                untracked: '',
                year: ''
            }
        }
        this.handleUserUpdate = handleUserUpdate.bind(this);
        this.handleDelete = handleDelete.bind(this);
        this.closeModal = closeModal.bind(this);
        this.openModal = openModal.bind(this);
        this.askIfSure = askIfSure.bind(this);
        this.openUpdateModal = openUpdateModal.bind(this);
        this.handleChange = handleChange.bind(this);
        this.toggleIncome = toggleIncome.bind(this);
        this.toggleUntracked = toggleUntracked.bind(this);
        this.handleCategoryUpdate = handleCategoryUpdate.bind(this);
    }

    render() {
        const props = this.props
        return (
            <div className="settings">
                <h2 className="settings">Change year view:</h2>
                <form className="settings">
                    <select
                        name="year_view"
                        defaultValue={props.current_user.year_view}
                    >
                        {props.current_user.years.map(year => <option value={year} key={year}>{year}</option>)}
                    </select>
                    <button className="year_view" onClick={this.handleUserUpdate}>Submit</button>
                </form>

                <h2 className="settings">Delete a Category:</h2>
                <form className="settings">
                    <select
                        defaultValue=""
                        name="delete_category"
                    >
                        <option value="" disabled>Select Category</option>
                        {!!props.current_user.categories ? props.current_user.categories.map(cat => <option value={cat.name} key={cat.id}>{cat.name}</option>) : null}
                    </select>
                    <button className="delete_category" type="button" onClick={this.askIfSure}>Delete</button>
                </form>

                <h2 className="settings">Update a Category:</h2>
                <form className="settings">
                    <select
                        defaultValue=""
                        name="update_category"
                    >
                        <option value="" disabled>Select Category</option>
                        {!!props.current_user.categories ? props.current_user.categories.map(cat => <option value={cat.name} key={cat.id}>{cat.name}</option>) : null}
                    </select>
                    <button className="update_category" type="button" onClick={this.openUpdateModal}>Update</button>
                </form>

                <AreYouSure
                    open={this.state.open}
                    onOpen={this.openModal}
                    closeModal={this.closeModal}
                    handleDelete={this.handleDelete}
                    message={"Are you sure you want to delete this category and all its entries?"}
                />
                <UpdateCategory
                    open={this.state.open}
                    onOpen={this.openModal}
                    closeModal={this.closeModal}
                    handleChange={this.handleChange}
                    toggleIncome={this.toggleIncome}
                    toggleUntracked={this.toggleUntracked}
                    handleUpdate={this.handleCategoryUpdate}
                    current_user={this.props.current_user}
                    name={this.state.form.name}
                    income={this.state.form.income}
                    untracked={this.state.form.untracked}
                    year={this.state.form.year}
                />
            </div>
        )
    }
}

const mapStateToProps = state => ({
    current_user: state.current_user
});
  
export default connect(mapStateToProps, { updateUser })(Settings);