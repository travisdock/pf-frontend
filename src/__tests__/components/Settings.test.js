import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Settings } from '../../components/Settings'
import * as SettingsHelpers from '../../components/helpers/settingsHelpers';
import * as ModalHelpers from '../../components/helpers/modalHelpers';

jest.mock('../../components/AreYouSure', () => () => 'AreYouSure')
jest.mock('../../components/UpdateCategory', () => () => 'UpdateCategory')

const props = {
    current_user: {
        user_id: 1,
        year_view: '2020',
        years: ['2018', '2019', '2020'],
        categories: [
            {
                id: 1,
                name: "categoryOne"
            },
            {
                id: 2,
                name: "categoryTwo"
            }
        ]
    }
};

describe('renders snapshots', () => {
    test('renders user <Settings />', () => {
        const component = renderer.create(
            <Settings {...props} />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('calls methods on button clicks', () => {
    test('handleUserUpdate is fired on button click', () => {
        const spy = jest.spyOn(SettingsHelpers, 'handleUserUpdate')
            .mockImplementationOnce(() => {});

        const component = shallow(<Settings {...props} />);

        const button = component.find('.year_view')
        button.simulate('click', { preventDefault () {} });

        expect(spy).toHaveBeenCalled();
    });

    test('askIfSure is fired on button click', () => {
        const spy = jest.spyOn(ModalHelpers, 'askIfSure')
            .mockImplementationOnce(() => {});

        const component = shallow(<Settings {...props} />);

        const button = component.find('.delete_category')
        button.simulate('click', { preventDefault () {} });

        expect(spy).toHaveBeenCalled();
    });

    test('openUpdateModal is fired on button click', () => {
        const spy = jest.spyOn(ModalHelpers, 'openUpdateModal')
            .mockImplementationOnce(() => {});

        const component = shallow(<Settings {...props} />);

        const button = component.find('.update_category')
        button.simulate('click', { preventDefault () {} });

        expect(spy).toHaveBeenCalled();
    });
});
