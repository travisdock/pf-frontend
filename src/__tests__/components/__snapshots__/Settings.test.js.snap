// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshots renders user <Settings /> 1`] = `
<div
  className="settings"
>
  <h2
    className="settings"
  >
    Change year view:
  </h2>
  <form
    className="settings"
  >
    <select
      defaultValue="2020"
      name="year_view"
    >
      <option
        value="2018"
      >
        2018
      </option>
      <option
        value="2019"
      >
        2019
      </option>
      <option
        value="2020"
      >
        2020
      </option>
    </select>
    <button
      className="year_view"
      onClick={[Function]}
    >
      Submit
    </button>
  </form>
  <h2
    className="settings"
  >
    Delete a Category:
  </h2>
  <form
    className="settings"
  >
    <select
      defaultValue=""
      name="delete_category"
    >
      <option
        disabled={true}
        value=""
      >
        Select Category
      </option>
      <option
        value="categoryOne"
      >
        categoryOne
      </option>
      <option
        value="categoryTwo"
      >
        categoryTwo
      </option>
    </select>
    <button
      className="delete_category"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </form>
  <h2
    className="settings"
  >
    Update a Category:
  </h2>
  <form
    className="settings"
  >
    <select
      defaultValue=""
      name="update_category"
    >
      <option
        disabled={true}
        value=""
      >
        Select Category
      </option>
      <option
        value="categoryOne"
      >
        categoryOne
      </option>
      <option
        value="categoryTwo"
      >
        categoryTwo
      </option>
    </select>
    <button
      className="update_category"
      onClick={[Function]}
      type="button"
    >
      Update
    </button>
  </form>
  AreYouSure
  UpdateCategory
</div>
`;

exports[`renders snapshots renders user <Settings /> without categories 1`] = `
<div
  className="settings"
>
  <h2
    className="settings"
  >
    Change year view:
  </h2>
  <form
    className="settings"
  >
    <select
      defaultValue="2020"
      name="year_view"
    />
    <button
      className="year_view"
      onClick={[Function]}
    >
      Submit
    </button>
  </form>
  <h2
    className="settings"
  >
    Delete a Category:
  </h2>
  <form
    className="settings"
  >
    <select
      defaultValue=""
      name="delete_category"
    >
      <option
        disabled={true}
        value=""
      >
        Select Category
      </option>
    </select>
    <button
      className="delete_category"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </form>
  <h2
    className="settings"
  >
    Update a Category:
  </h2>
  <form
    className="settings"
  >
    <select
      defaultValue=""
      name="update_category"
    >
      <option
        disabled={true}
        value=""
      >
        Select Category
      </option>
    </select>
    <button
      className="update_category"
      onClick={[Function]}
      type="button"
    >
      Update
    </button>
  </form>
  AreYouSure
  UpdateCategory
</div>
`;
